cpp/WKTJsiWorkletContext.cpp

```diff
@@ -219,8 +219,9 @@ JsiWorkletContext::createCallInContext(jsi::Runtime &runtime,
     // Start by wrapping the arguments
     ArgumentsWrapper argsWrapper(runtime, arguments, count);
 
-    // Wrap the this value
-    auto thisWrapper = JsiWrapper::wrap(runtime, thisValue);
+    // In hermes-engine, "thisValue" appears to always be an undefined value, but in jsc it's the globalThis object...
+    // It's just easier to wrap an undefined and provide that as the "this" context.
+    auto thisWrapper = JsiWrapper::wrap(runtime, jsi::Value::undefined());
```

Comment:
Hi! Thanks for the PR!! I'm pretty happy with the native state parts of this PR, but not 100% in favour of the above change, since the point behind keeping the this-object is to offer a predictable way to call functions across runtimes. It should be legal to wrap `jsi::Value::undefined()` so maybe this fix is not needed?

Reply:
The problem exists when in JSC, where `thisValue` is *not* undefined. I was just being lazy and making it always undefined to match what hermes appears to do. I'm happy to split the PR and deal with the undefined portion separately.

Reply:
The `EXC_BAD_ACCESS` issue from wrapping globalThis can be seen very quickly by just running the example/test harness app with hermes disabled, then running the "nested worklet" test case.